lib/replicator.js

```diff
@@ -2086,16 +2086,27 @@ module.exports = class Replicator {
     this._maybeResolveIfAvailableRanges()
   }
 
-  _closeSession (session) {
-    session.close().catch(safetyCatch)
+  _closeSession () {
+    this.core.active--
+    if (this.core.sessions.length === 1 && this.core.active === 1 && this.core.sessions[0].autoClose) {
+      this.core.sessions[0].close().catch(safetyCatch)
+    }
+    if (this.core.active === 0 && this.core.sessions.length === 0) {
```

Comment:
Could benefit from a comment; because I'm 90% sure it will take me a while to understand this code next time I read it. Something like

`// the underlying session already closed before, and this is the last replication session. So the core should be closed now.